homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py

```diff
@@ -0,0 +1,211 @@
+        )
+
+        # We also needs to execute these 3 commands to make it work correctly
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_DEROGATION_ON_OFF_STATE, OverkizCommandParam.OFF
+        )
+        await self.executor.async_execute_command(
+            OverkizCommand.REFRESH_PASS_APC_HEATING_MODE
+        )
+        await self.executor.async_execute_command(
+            OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE
+        )
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode, e.g., home, away, temp."""
+        heating_mode = cast(
+            str, self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_MODE)
+        )
+
+        if heating_mode == OverkizCommandParam.INTERNAL_SCHEDULING:
+            # In Internal scheduling, it could be comfort or eco
+            return OVERKIZ_TO_PROFILE_MODES[
+                cast(
+                    str,
+                    self.executor.select_state(
+                        OverkizState.IO_PASS_APC_HEATING_PROFILE
+                    ),
+                )
+            ]
+
+        return OVERKIZ_TO_PRESET_MODES[heating_mode]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_PASS_APC_HEATING_MODE,
+            PRESET_MODES_TO_OVERKIZ[preset_mode],
+        )
+
+        # We also needs to execute these 3 commands to make it work correctly
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_DEROGATION_ON_OFF_STATE, OverkizCommandParam.OFF
+        )
+        await self.executor.async_execute_command(
+            OverkizCommand.REFRESH_PASS_APC_HEATING_MODE
+        )
+        await self.executor.async_execute_command(
+            OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return hvac target temperature."""
+        current_profile = cast(
+            str,
+            self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_PROFILE),
+        )
+        if current_profile == OverkizCommandParam.ECO:
+            return cast(
+                float,
+                self.executor.select_state(
+                    OverkizState.CORE_ECO_HEATING_TARGET_TEMPERATURE
+                ),
+            )
+        if current_profile == OverkizCommandParam.COMFORT:
+            return cast(
+                float,
+                self.executor.select_state(
+                    OverkizState.CORE_COMFORT_HEATING_TARGET_TEMPERATURE
+                ),
+            )
+        if current_profile == OverkizCommandParam.DEROGATION:
+            return cast(
+                float,
+                self.executor.select_state(
+                    OverkizState.CORE_DEROGATED_TARGET_TEMPERATURE
+                ),
+            )
```

Comment:
Perhaps you can use a map here, like we do for the others? In this case we only check for a string in current_profile and map it to a state key.

Reply:
Done.